src/components/slider/slider-comparator.vue

```diff
@@ -0,0 +1,192 @@
+    if (topLayer !== oldTopLayer) {
+      deactivate()
+    }
+
+    activate()
+  } else {
+    deactivate()
+  }
+
+  olMap.value?.render()
+})
+
+watch(sliderOffset, sliderOffset => {
+  styleObject.left = `${sliderOffset}px`
+  olMap.value?.render()
+})
+
+function activate() {
+  const olLayer = openLayers.getLayerFromCache(sliderTopLayer.value)
+
+  olLayerPrerenderEvent = olLayer.on(
+    <EventTypes>EventType.PRERENDER,
+    function (event) {
+      const ctx = <CanvasRenderingContext2D>(<RenderEvent>event).context
+      const mapSize = olMap.value?.getSize()!
+      const width = sliderOffset.value + sliderElement.value!.offsetWidth / 2
+      const tl = getRenderPixel(<RenderEvent>event, [width, 0])
+      const tr = getRenderPixel(<RenderEvent>event, [mapSize[0], 0])
+      const bl = getRenderPixel(<RenderEvent>event, [width, mapSize[1]])
+      const br = getRenderPixel(<RenderEvent>event, mapSize)
+
+      ctx.save()
+      ctx.beginPath()
+      ctx.moveTo(tl[0], tl[1])
+      ctx.lineTo(bl[0], bl[1])
+      ctx.lineTo(br[0], br[1])
+      ctx.lineTo(tr[0], tr[1])
+      ctx.closePath()
+      ctx.clip()
+    }
+  )
+
+  olLayerPostrenderEvent = olLayer.on(
+    <EventTypes>EventType.POSTRENDER,
+    function (e) {
+      const event = <RenderEvent>e
+      const ctx = <CanvasRenderingContext2D>event.context
+      ctx.restore()
+    }
+  )
+}
+
+function deactivate() {
+  unByKey([olLayerPrerenderEvent, olLayerPostrenderEvent])
+}
+
+function moveSplitBar(offsetLeft: number) {
+  const mapSize = olMap.value?.getSize()!
+  const newRatio =
+    (offsetLeft + sliderElement.value!.offsetWidth / 2) / mapSize[0]
+
+  sliderStore.setRatio(newRatio)
+}
+
+function onMouseDown() {
+  isDragging = true
+}
+
+function onMouseMove(payload: MouseEvent) {
+  if (!isDragging) {
+    return
+  }
+
+  const mapContainerOffsetLeft = mapWrapperElement.offsetLeft!
+
+  moveSplitBar(payload.clientX - mapContainerOffsetLeft)
+}
+
+function onMouseUp() {
+  isDragging = false
+}
+
+function onKeyDownRight() {
+  moveSplitBar(sliderElement.value!.offsetLeft + DEFAULT_STEP_ONKEYDOWN)
+}
+
+function onKeyDownLeft() {
+  moveSplitBar(sliderElement.value!.offsetLeft - DEFAULT_STEP_ONKEYDOWN)
+}
+
+function onKeyDownEsc() {
+  sliderStore.toggleSlider()
+}
+
+onMounted(() => {
+  // Keep ".map-wrapper" for v3 compatibility
+  mapWrapperElement = olMap.value
+    ?.getTargetElement()
+    ?.closest('.map-wrapper') as HTMLElement
+
+  document.addEventListener('mousemove', onMouseMove)
+  document.addEventListener('mouseup', onMouseUp)
+})
+
+onUnmounted(() => {
+  document.addEventListener('mousemove', onMouseMove)
+  document.removeEventListener('mouseup', onMouseUp)
+
+  // Unbind event for HMR support
+  deactivate()
+})
+</script>
+
+<template>
+  <button
+    ref="sliderElement"
+    v-if="sliderActive"
+    @mousedown="onMouseDown"
+    @mousemove="onMouseMove"
+    @mouseup="onMouseUp"
+    @keydown.space="onKeyDownRight"
```

Comment:
if we have "space", there should be backspace (?) for opposite movement, but this is used for backwards navigation in browser history => maybe shift + space for left movement ??

Reply:
Thx for the suggestion => added backspace handler.